### PR TITLE
FIX - allow datasets without headshape dig to be coregistered in report

### PR DESCRIPTION
**Pull request by jstout211**
#### What does this implement/fix?
If data is acquired without a headshape, the report will error during the coregistration.  The distance measurement calculation will fail because the headshape can't be evaluated.

#### Additional information
Our lab currently does not collect a headshape, but rather uses localized fiducials with vitamin E capsules on the MRI.  Running data through the mne-bids-pipeline fails during report generation.

### CODE REVIEW

File: mne/report/report.py [hunk 1]
```diff
@@ -490,14 +490,17 @@ def _itv(function, fig, **kwargs):
...
+    try:
+        dists = dig_mri_distances(info=kwargs['info'],
+                                  trans=kwargs['trans'],
+                                  subject=kwargs['subject'],
+                                  subjects_dir=kwargs['subjects_dir'])
+        caption = (f'Average distance from {len(dists)} digitized points to '
+                   f'head: {1e3 * np.mean(dists):.2f} mm')
+    except BaseException as e:
+        caption = ('Distances could not be calculated from digitized points')
+        logger.warning(e)
```
Comment (larsoner) [inline mne/report/report.py:502]: Instead of `logger.warning` this should be `warn(...)` from `..utils`. It does smart emission based on `verbose` level

**Comment by larsoner** [comment]
Can you add a quick test that this works? I think something like this would do it:
```
with info._unlock():
    dig, info['dig'] = info['dig'], []
with pytest.warns(RuntimeWarning, match='could not be calculated'):
    report.add_...
with info._unlock():
    info['dig'] = dig
```
Also there is a conflict that you need to fix via rebase or merge.

If it's not clear or you don't want to look into these things, I can push a quick commit

**Comment by agramfort** [comment]
@jstout211 you need to fix conflicts here

**Comment by jstout211** [comment]
@larsoner  @agramfort  - I had to pull in updates of report.py on main that were pushed today.  I think this should solve it.  

I am not familiar enough with the report code to make a test for it.

**Comment by larsoner** [comment]
> I am not familiar enough with the report code to make a test for it.

@jstout211 feel free to look at  dbb8c92 to see how I added a version of the suggested code above.

FYI I got an error with your code when I added a test for it (caused by the switch to `warn` from `logger.warning` I think):
```
mne/report/report.py:461: in _iterate_trans_views
    return _itv(function, fig, surfaces=['head-dense'], **kwargs)
mne/report/report.py:503: in _itv
    warn(e)
mne/utils/_logging.py:374: in warn
    warnings.warn_explicit(
E   TypeError: expected string or bytes-like object
```
A cast to string fixed it (explicitly done now via f-string).

**Review by larsoner — APPROVED** [review]
@hoechenberger feel free to merge if you're happy and CIs come back green

File: mne/report/report.py [hunk 2]
```diff
@@ -500,7 +503,7 @@ def _itv(function, fig, **kwargs):
                    f'head: {1e3 * np.mean(dists):.2f} mm')
     except BaseException as e:
         caption = ('Distances could not be calculated from digitized points')
```
Comment (drammock) [inline mne/report/report.py:505]:
```suggestion
        caption = 'Distances could not be calculated from digitized points'
```

**Review by drammock — APPROVED** [review]
LGTM, just one cosmetic suggestion.

**Comment by hoechenberger** [comment]
Thanks, @jstout211!

**Comment by hoechenberger** [comment]
We forgot to add a changelog entry 😱

**Pull request merged**